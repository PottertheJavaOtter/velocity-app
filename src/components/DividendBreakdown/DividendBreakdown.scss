.DividendBreakdown {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;

  .DividendBreakdown__columns {
    align-items: center;
    display: flex;
    justify-content: space-evenly;
    width: 100%;
  }
}
