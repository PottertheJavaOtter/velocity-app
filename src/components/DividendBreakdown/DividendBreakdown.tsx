import React from "react";
import { IonText } from "@ionic/react";
import { formatDecimalToDollars } from "../../util/moneyFormatter";

import "./DividendBreakdown.scss";

interface DividendBreakdownProps {
  /** The total annual dividend */
  dividend: number;
}
const DividendBreakdown: React.FC<DividendBreakdownProps> = ({ dividend }) => {
  const annualDividend = formatDecimalToDollars(dividend);
  const quarterlyDividend = formatDecimalToDollars(dividend / 4);
  const dailyDividend = formatDecimalToDollars(dividend / 12);

  return (
    <section className="DividendBreakdown">
      <IonText color="dark">
        <h5>Annually</h5>
        <p>{annualDividend}</p>
      </IonText>
      <div className="DividendBreakdown__columns">
        <IonText color="dark">
          <h5>Quarterly</h5>
          <p>{quarterlyDividend}</p>
        </IonText>
        <IonText color="dark">
          <h5>Monthly</h5>
          <p>{dailyDividend}</p>
        </IonText>
      </div>
    </section>
  );
};

export default DividendBreakdown;
