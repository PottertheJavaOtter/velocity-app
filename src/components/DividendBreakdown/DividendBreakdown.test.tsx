import React from "react";
import { render } from "@testing-library/react";
import DividendBreakdown from "./DividendBreakdown";

describe("DividendBreakdown", () => {
  it("should render the dividend breakdowns", () => {
    const { container } = render(<DividendBreakdown dividend={10909.12} />);
    expect(container).toMatchSnapshot();
  });
});
