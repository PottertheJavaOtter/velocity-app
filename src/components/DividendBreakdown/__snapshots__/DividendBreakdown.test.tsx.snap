// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DividendBreakdown should handle 0 dividend 1`] = `
<div>
  <section
    class="DividendBreakdown"
  >
    <ion-text
      color="dark"
    >
      <h5>
        Annually
      </h5>
      <p>
        $0.00
      </p>
    </ion-text>
    <div
      class="DividendBreakdown__columns"
    >
      <ion-text
        color="dark"
      >
        <h5>
          Quarterly
        </h5>
        <p>
          $0.00
        </p>
      </ion-text>
      <ion-text
        color="dark"
      >
        <h5>
          Monthly
        </h5>
        <p>
          $0.00
        </p>
      </ion-text>
    </div>
  </section>
</div>
`;

exports[`DividendBreakdown should render the dividend breakdowns 1`] = `
<div>
  <section
    class="DividendBreakdown"
  >
    <ion-text
      color="dark"
    >
      <h5>
        Annually
      </h5>
      <p>
        $10,909.12
      </p>
    </ion-text>
    <div
      class="DividendBreakdown__columns"
    >
      <ion-text
        color="dark"
      >
        <h5>
          Quarterly
        </h5>
        <p>
          $2,727.28
        </p>
      </ion-text>
      <ion-text
        color="dark"
      >
        <h5>
          Monthly
        </h5>
        <p>
          $909.09
        </p>
      </ion-text>
    </div>
  </section>
</div>
`;
